packages/core-plugin-api/src/extensions/extensions.tsx

```diff
@@ -161,9 +161,14 @@ export function createComponentExtension<
    * variable for this extension.
    */
   name?: string;
+
+  /**
+   * Additional data to attach to the component.
+   */
+  data?: Record<string, any>;
```

Comment:
```suggestion
  data?: Record<string, unknown>;
```

to align with the `createReactExtension` type

Reply:
Ah true, will fix!